packages/shared/src/uri/Uri.ts

```diff
@@ -0,0 +1,78 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import type { Uri as APIUri } from '@podman-desktop/api';
+
+export class Uri {
```

Comment:
Could you explain why we need to have our own implementation of the Uri class ? I am not sure to get it

Reply:
The Uri object which is returned when calling the api is serialized using private properties bc the Uri constructor uses `private _scheme ..., private _path ...`. So when we receive the resulting object in ai lab we have 
```
_path: '...',
_scheme: '...',
```
In this way it is not kinda usable so we need to "revive" it so that its fields are exposed through methods like the original Uri.
The same thing is done on Desktop.

Reply:
What are the properties that we need ?